MaxText/layers/linears.py

```diff
@@ -192,5 +238,11 @@ def __call__(self, inputs, decode: bool = False, deterministic: bool = False):
         kernel_axes=('mlp', 'embed'),
         name='wo',
         use_int8=cfg.int8_training,
+        use_bias=self.use_bias,
     )(x)
+
+    if self.apply_padding_mask and padding_mask is not None:
+      output *= padding_mask
+    if self.add_skip_connection:
+      output += residual
```

Comment:
I don't understand why we need to push add_skip_connection in MLPBlock? Can't the caller own the residual layer (if they want it)?

Reply:
Generally, anything that is easy to control from user code should get pushed into there -- there shared building blocks should aim to do the minimum.

Reply:
Gotcha, thank you for the advice, I think they are equivalent as well. I followed [the implementation from praxis](https://github.com/google/praxis/blob/fbde0b77ff4a77fb12f05ffaf2adce9a4c7806ed/praxis/layers/transformers.py#L467) and hope to get aligned for numerical verification per each module.

Reply:
You can put your numerical checks elsewhere. That isn't a reason to create additional complexity!

Reply:
Moved skip_connection in decoder layer.